eng/targets/Java.Common.props

```diff
@@ -0,0 +1,9 @@
+<Project>
+  <PropertyGroup>
+    <!-- Disable gradle daemon on CI since the CI seems to try to wait for the daemon to shut down, which it doesn't do -->
+    <GradleOptions Condition="'$(ContinuousIntegrationBuild)' == 'true'">$(GradleOptions) -Dorg.gradle.daemon=false</GradleOptions>
+    <Configuration Condition="'$(Configuration)' == '' AND '$(ContinuousIntegrationBuild)' == 'true'">Release</Configuration>
+    <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
```

Comment:
Why is this needed❔ `$(Configuration)` should always be set at least when using our `build.*` scripts.